### PR TITLE
NIFI-6513 PutHive3Streaming.AUTO_CREATE_PARTITIONS is not used	

### PR DESCRIPTION
**Pull request by adarmiento**
#### Description of PR

- Removed unused `AUTOCREATE_PARTITIONS` from `PutHive3Streaming`
 - Renamed `PARTITION_VALUES` to `STATIC_PARTITION_VALUES` for correctness and better understanding
- `STATIC_PARTITION_VALUES` descriptions clearly states that having that property filler implies Hive Static Partitioning

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [x] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by AxelSync** [comment]
cc @mattyb149 could you please review this PR? 
Thanks

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/processors/hive/PutHive3Streaming.java [hunk 1]
```diff
@@ -151,28 +151,19 @@
...
+    static final PropertyDescriptor STATIC_PARTITION_VALUES = new PropertyDescriptor.Builder()
             .name("hive3-stream-part-vals")
-            .displayName("Partition Values")
+            .displayName("Static Partition Values")
             .description("Specifies a comma-separated list of the values for the partition columns of the target table. If the incoming records all have the same values "
                     + "for the partition columns, those values can be entered here, resulting in a performance gain. If specified, this property will often contain "
                     + "Expression Language, for example if PartitionRecord is upstream and two partitions 'name' and 'age' are used, then this property can be set to "
-                    + "${name},${age}.")
+                    + "${name},${age}.\n"
+                    + "Having this property filled implies Hive Static Partitioning.")
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/processors/hive/PutHive3Streaming.java]: We should elaborate on this for clarity, as well as updating the `CapabilityDescription` above to say that if `Static Partition Values` is set, the partition columns (if present in the record) are ignored, and if `Static Partition Values` is not set, the partition columns must be declared at the end of the record in the order they were declared in the table definition. We can add similar text to both the Capability Description and this property, although the property doc can be less verbose.

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/processors/hive/PutHive3Streaming.java [hunk 2]
```diff
@@ -151,28 +151,19 @@
...
             .required(false)
             .expressionLanguageSupported(ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
             .build();
 
-    static final PropertyDescriptor AUTOCREATE_PARTITIONS = new PropertyDescriptor.Builder()
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/processors/hive/PutHive3Streaming.java]: Removing a required field will cause all existing processors to be marked invalid upon upgrade. Not saying we should keep it, but wanted to leave that note here for posterity :)

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/processors/hive/PutHive3Streaming.java (hunk 1 above)
Comment (adarmiento) [reply nifi-nar-bundles/nifi-hive-bundle/nifi-hive3-processors/src/main/java/org/apache/nifi/processors/hive/PutHive3Streaming.java]: Hello, and sorry for the late reply. 

Thanks for the update and the tips. I'm working on it

**Comment by adarmiento** [comment]
Hello @mattyb149 
Do you think the additional doc is enough? Thanks

**Comment by adarmiento** [comment]
Ping @mattyb149 any news about this one? 
Thanks

**Comment by mattyb149** [comment]
+1 LGTM although I added more doc to the `Static Partition Values` description and the overall CapabilityDescription for the processor for clarity. Ran with and without static values supplied, everything ran well. Thanks for this improvement! Merging to master

**Pull request closed without merging**